src/lib/WriteThroughCache.ts

```diff
@@ -81,35 +86,37 @@ export class WriteThroughCache extends CacheInstance {
   public async delValue(key: string): Promise<void> {
     this.emit('del', key);
     await this.localCache.delValue(key);
-    await this.redisCache.delValue(key);
+    await this.redisCacheForWriting.delValue(key);
   }
 
   /**
    * @inheritdoc
    */
   public async clear(): Promise<void> {
     await this.localCache.clear();
-    await this.redisCache.clear();
+    await this.redisCacheForWriting.clear();
+    await this.redisCacheForReading.clear();
   }
 
   /**
    * @inheritdoc
    */
   public async clearMemory(): Promise<void> {
     await this.localCache.clearMemory();
-    await this.redisCache.clearMemory();
+    await this.redisCacheForWriting.clearMemory();
+    await this.redisCacheForReading.clearMemory();
   }
 
   public isLockingSupported(): boolean {
     return true;
   }
 
   public lock(resource: string, ttlMs: number): Promise<any> {
-    return this.redisCache.lock(resource, ttlMs);
+    return this.redisCacheForWriting.lock(resource, ttlMs) && this.redisCacheForReading.lock(resource, ttlMs) ;
```

Comment:
```suggestion
    return this.redisCacheForWriting.lock(resource, ttlMs);
```
In which case do we have to lock Redis? Can we lock only the writing node?

Reply:
we don't actually, it is not supported anymore

```js
  public lock(resource: string, ttlMs: number): Promise<any> {
    throw new Error('unsupported');
  }
```

Reply:
@AdrienDesola actually we have to lock/unlock only the primary instance